plugins/rocketmq/src/main/java/com/navercorp/pinpoint/plugin/rocketmq/RocketMQPlugin.java

```diff
@@ -0,0 +1,200 @@
+            if (updateNameServerAddressListMethod != null) {
+                updateNameServerAddressListMethod.addInterceptor(UpdateNameServerAddressListInterceptor.class);
+            }
+
+            return target.toBytecode();
+        }
+    }
+
+    public static class SendCallbackTransform implements TransformCallback {
+
+        @Override
+        public byte[] doInTransform(Instrumentor instrumentor, ClassLoader classLoader, String className,
+                                    Class<?> classBeingRedefined, ProtectionDomain protectionDomain,
+                                    byte[] classfileBuffer) throws InstrumentException {
+            final InstrumentClass target = instrumentor.getInstrumentClass(classLoader, className,
+                                                                           classfileBuffer);
+            target.addField(AsyncContextAccessor.class);
+            final List<InstrumentMethod> constructors = target.getDeclaredConstructors();
+            for (InstrumentMethod constructor : constructors) {
+                constructor.addInterceptor(ConstructInterceptor.class);
+            }
+            final MethodFilter onSuccessFilter = MethodFilters.chain(
+                    MethodFilters.name("onSuccess"),
+                    MethodFilters.argAt(0, "org.apache.rocketmq.client.producer.SendResult"));
+            final List<InstrumentMethod> onSuccessMethods = target.getDeclaredMethods(onSuccessFilter);
+            for (InstrumentMethod instrumentMethod : onSuccessMethods) {
+                instrumentMethod.addScopedInterceptor(OnSuccessInterceptor.class, RocketMQConstants.SCOPE);
+            }
+
+            final MethodFilter onExceptionFilter = MethodFilters.chain(
+                    MethodFilters.name("onException"),
+                    MethodFilters.argAt(0, "java.lang.Throwable"));
+            final List<InstrumentMethod> onExceptions = target.getDeclaredMethods(onExceptionFilter);
+            for (InstrumentMethod instrumentMethod : onExceptions) {
+                instrumentMethod.addScopedInterceptor(OnExceptionInterceptor.class, RocketMQConstants.SCOPE);
+
+            }
+            return target.toBytecode();
+        }
+    }
+
+    public static class MessageListenerConcurrentlyTransform implements TransformCallback {
+
+        @Override
+        public byte[] doInTransform(Instrumentor instrumentor, ClassLoader classLoader, String className,
+                                    Class<?> classBeingRedefined, ProtectionDomain protectionDomain,
+                                    byte[] classfileBuffer) throws InstrumentException {
+            final InstrumentClass target = instrumentor.getInstrumentClass(classLoader, className,
+                                                                           classfileBuffer);
+            final List<InstrumentMethod> consumeMessageMethods = target.getDeclaredMethods(
+                    MethodFilters.name("consumeMessage"));
```

Comment:
In my opinion, it would be better if you don't use MethodFilter like this.
In this case, it must execute the method declared in the interface.

In general, when there is a Boundary level scope, it is a case of many overloading.
This is because the user does not know which method to call.

Like below.
```
public void invoke() {
invoke("hello");
}

public void invoke(String message) {
invoke(message, null);
}

public void invoke(String message1, String message2) {
// todo
}
```

However, in this case, there is no need to do this because there are methods that must pass through the interface.

And, If you don't use it, you can reduce complex and you can reduce calculation time.

Reply:
> In my opinion, it would be better if you don't use MethodFilter like this.
> In this case, it must execute the method declared in the interface.
> 
> In general, when there is a Boundary level scope, it is a case of many overloading.
> This is because the user does not know which method to call.
> 
> Like below.
> 
> ```
> public void invoke() {
> invoke("hello");
> }
> 
> public void invoke(String message) {
> invoke(message, null);
> }
> 
> public void invoke(String message1, String message2) {
> // todo
> }
> ```
> 
> However, in this case, there is no need to do this because there are methods that must pass through the interface.
> 
> And, If you don't use it, you can reduce complex and you can reduce calculation time.

Fix it.
BTW,when intercept an internal interface, it should use the `InterfaceInternalNameMatcherOperand `and must have a `basepackage`,like [this](https://github.com/messi-gao/pinpoint/blob/feature/rocketmq/plugins/rocketmq/src/main/java/com/navercorp/pinpoint/plugin/rocketmq/RocketMQPlugin.java#L75).is there a way to avoid pass the `basepackage `param?

Reply:
I am not familiar with RocketMQ. 
So, I am going to answer it, after I research RockerMQ structure. :)